### PR TITLE
Fix for rustc 1.18.0-nightly (5c94997b6 2017-03-30)

### PR DESCRIPTION
**Pull request by ensch**
The codes compiles, but does not work.

### CODE REVIEW

File: clippy_lints/src/copies.rs [hunk 1]
```diff
@@ -225,9 +225,11 @@ fn if_sequence(mut expr: &Expr) -> (SmallVector<&Expr>, SmallVector<&Block>) {
     let mut conds = SmallVector::new();
     let mut blocks = SmallVector::new();
 
-    while let ExprIf(ref cond, ref then_block, ref else_expr) = expr.node {
+    while let ExprIf(ref cond, ref then_expr, ref else_expr) = expr.node {
         conds.push(&**cond);
-        blocks.push(&**then_block);
+        //FIXME
+        //blocks.push(&**then_expr);
```
Comment (oli-obk) [inline clippy_lints/src/copies.rs]: You need to change the return type to Expr instead of block. Then it'll work again.

Not sure about the use sites oft this function. Might need to match on the kind

File: clippy_lints/src/entry.rs [hunk 2]
```diff
@@ -46,8 +46,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for HashMapLint {
                 if let Some((ty, map, key)) = check_cond(cx, check) {
                     // in case of `if !m.contains_key(&k) { m.insert(k, v); }`
                     // we can give a better error message
-                    let sole_expr = else_block.is_none() &&
-                                    ((then_block.expr.is_some() as usize) + then_block.stmts.len() == 1);
```
Comment (oli-obk) [inline clippy_lints/src/entry.rs]: You need to match on then_block.node to extract the block, if it is one

File: clippy_lints/src/let_if_seq.rs [hunk 3]
```diff
@@ -69,10 +69,9 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for LetIfSeq {
                 let hir::StmtExpr(ref if_, _) = expr.node,
                 let hir::ExprIf(ref cond, ref then, ref else_) = if_.node,
                 !used_in_expr(cx, def_id, cond),
-                let Some(value) = check_assign(cx, def_id, then),
```
Comment (oli-obk) [inline clippy_lints/src/let_if_seq.rs]: Undo the changes and add let ExprBlock(then) = then.node,

File: clippy_lints/src/let_if_seq.rs [hunk 4]
```diff
@@ -105,9 +104,9 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for LetIfSeq {
                     mut=mutability,
                     name=name.node,
                     cond=snippet(cx, cond.span, "_"),
-                    then=if then.stmts.len() > 1 { " ..;" } else { "" },
```
Comment (oli-obk) [inline clippy_lints/src/let_if_seq.rs]: Undo

**Comment by oli-obk** [comment]
Thanks so much for doing this!

**Comment by dtolnay** [comment]
@ensch are you still working on addressing the build failure?

**Comment by ensch** [comment]
No, I am not.

**Comment by oli-obk** [comment]
I probably won't get to this before monday.

**Comment by oli-obk** [comment]
```rust
let monday = now;
```

Probability is a great thing.

I'll take this now

**Comment by oli-obk** [comment]
the `"{}"` in `panic!("{}")` does no longer have any expansion info whatsoever... This broke the lint checking for such mistakes. I could not figure out how to fix it.

**Comment by oli-obk** [comment]
@llogiq @mcarton are you alright with publishing a new version while the lint is broken (you two created and heavily extended the lint). Broken meaning it doesn't trigger at all. Alternatively broken can mean a single false positive for which we already have a closed issue.

**Comment by llogiq** [comment]
Yeah, let's publish, reopen the issue while we fix it and let people update already.
